src/osdc/Objecter.cc

```diff
@@ -870,18 +871,18 @@ void Objecter::_linger_submit(LingerOp *info, shunique_lock& sul)
   _send_linger(info, sul);
 }
 
-struct C_DoWatchNotify : public Context {
-  Objecter *objecter;
-  Objecter::LingerOp *info;
-  MWatchNotify *msg;
-  C_DoWatchNotify(Objecter *o, Objecter::LingerOp *i, MWatchNotify *m)
-    : objecter(o), info(i), msg(m) {
-    info->get();
+class CB_DoWatchNotify {
+  Objecter* objecter;
+  intrusive_ptr<Objecter::LingerOp> info;
```

Comment:
should not release `info` and `msg` in the dtor of `CB_DoWatchNotify`

Reply:
@adamemerson by using the raw pointer and only do get() in the ctor, the `ceph_test_rados_api_watch_notify` test passes.